tests/rados/test_bluestore_configs.py

```diff
@@ -109,3 +111,108 @@ def create_pool_write_iops(param, pool_type):
+            "\n\t 2. Modify the value of bluestore_cache_size_ssd and bluestore_cache_size_hdd"
+            "\n\t 3. Verify the values being reflected in ceph config"
+            "\n\t 4. Create replicated and ec pool and perform IOPS"
+            "\n\t 5. cleanup - Remove all the pools created and reset configs modified"
+        )
+        log.info(doc)
+        log.info("Running test case to verify BlueStore Cache size tuning")
+
+        try:
+            # verify default value for bluestore cache
+            out = mon_obj.get_config(section="osd", param="bluestore_cache_size")
+            log.info(f"bluestore_cache_size default value - {out} | Expected 0")
+            assert int(out.strip("\n")) == 0
+
+            out = mon_obj.get_config(section="osd", param="bluestore_cache_size_hdd")
+            log.info(
+                f"bluestore_cache_size_hdd default value - {out} | Expected 1073741824"
+            )
+            assert int(out.strip("\n")) == 1073741824
+
+            out = mon_obj.get_config(section="osd", param="bluestore_cache_size_ssd")
+            log.info(
+                f"bluestore_cache_size_ssd default value - {out} | Expected 3221225472"
+            )
+            assert int(out.strip("\n")) == 3221225472
+
+            def modify_cache_size(factor):
+                cache_value = int(1073741824 * factor)
+                cache_cfg = {
+                    "section": "osd",
+                    "name": "bluestore_cache_size_hdd",
+                    "value": cache_value,
+                }
+                assert mon_obj.set_config(**cache_cfg)
+                out = mon_obj.get_config(
+                    section="osd", param="bluestore_cache_size_hdd"
+                )
+                log.info(
+                    f"bluestore_cache_size_hdd modified value - {out} | Expected {cache_value}"
+                )
+                assert int(out.strip("\n")) == cache_value
+
+                cache_value = int(3221225472 * factor)
+                cache_cfg = {
+                    "section": "osd",
+                    "name": "bluestore_cache_size_ssd",
+                    "value": cache_value,
+                }
+                assert mon_obj.set_config(**cache_cfg)
+                out = mon_obj.get_config(
+                    section="osd", param="bluestore_cache_size_ssd"
+                )
+                log.info(
+                    f"bluestore_cache_size_ssd modified value - {out} | Expected {cache_value}"
+                )
+                assert int(out.strip("\n")) == cache_value
+
+            # modify ssd and hdd cache (increase)
+            modify_cache_size(factor=1.5)
+
+            # restart osd services
+            restart_osd_service()
+
+            # perform iops
+            create_pool_write_iops(param="cache_inc", pool_type="replicated")
+            create_pool_write_iops(param="cache_inc", pool_type="ec")
+
+            # modify ssd and hdd cache (decrease)
```

Comment:
All the OSD's configured on HDDs, not on SSDs.

Reply:
Yes, I understand, but we are still modifying the parameter to see if the value is getting changed correctly.
On rhos-d VM clusters, all the pools created will only consider the value being set in `bluestore_cache_size_hdd`